apps/voting-disputable/contracts/DisputableVoting.sol

```diff
@@ -1022,8 +1061,10 @@ contract DisputableVoting is IForwarderWithContext, DisputableAragonApp {
     * @param _voter Address of principal voter
     * @param _caster Address of vote caster, if voting via representative
     */
+    // Note: not sure about pushing these to be private 🤷‍♂️
     function _castVote(Vote storage _vote, uint256 _voteId, bool _supports, address _voter, address _caster) private {
         Setting storage setting = settings[_vote.settingId];
+        // Note: if we move the quiet ending check at the end here, we could optimize this so it doesn't get calculated outside of the quiet ending period
```

Comment:
I think this and the next note are important, and related to the issue we have with `_hasEnded()`. It would be ideal if we updated the quiet ending state as a modifier / precondition to the rest of functions, so that we effectively work with the latest state down the line.

Reply:
Having it here allows us to work with the previous and current support values to now if the outcome was flipped actually. I think having this as a precondition is less intuitive, don't you think? I mean, by definition is something you would always like to check after a vote, not before

Reply:
> I mean, by definition is something you would always like to check after a vote, not before

But from what I understand, not really, right? We're actually using this as a pre-condition because we only know about a quiet ending extension when someone votes (or tries to close) in the new period.

Reply:
Addressed in 798589a04a663ab217c717e52a6f73c81a3dc196